Okta/OktaNetworkRequestCustomizationDelegate.h

```diff
@@ -0,0 +1,26 @@
+/*! @file OktaNetworkRequestCustomizationDelegate.h
+   @brief AppAuth iOS SDK
+   @copyright
+       Copyright 2015 Google Inc. All Rights Reserved.
+   @copydetails
+       Licensed under the Apache License, Version 2.0 (the "License");
+       you may not use this file except in compliance with the License.
+       You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+       Unless required by applicable law or agreed to in writing, software
+       distributed under the License is distributed on an "AS IS" BASIS,
+       WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+       See the License for the specific language governing permissions and
+       limitations under the License.
+*/
+
+#import <Foundation/Foundation.h>
+
+@protocol OktaNetworkRequestCustomizationDelegate <NSObject>
+
+- (nullable NSURLRequest*)sendCustomizableRequest: (nullable NSURLRequest *)request;
```

Comment:
please rename this to something like customizeRequest or modifyRequest, it does not send a request.
@stevelind-okta @IldarAbdullin-okta  any preferences here?

Reply:
`customizableURLRequest `